src/ceph-volume/ceph_volume/tests/test_terminal.py

```diff
@@ -66,3 +71,51 @@ def test_subcommand_found_and_dispatched_with_errors(self):
         with pytest.raises(SystemExit) as error:
             terminal.dispatch({'sub': BadSubCommand}, argv=['sub'])
         assert str(error.value) == '100'
+
+
+@pytest.fixture
+def stream(buffer, encoding):
+    # mock a stdout with given encoding
+    if sys.version_info >= (3, 0):
+        stdout = sys.stdout
+        stream = io.TextIOWrapper(buffer,
+                                  encoding=encoding,
+                                  errors=stdout.errors,
+                                  newline=stdout.newlines,
+                                  line_buffering=stdout.line_buffering)
+    else:
+        stream = codecs.getwriter(encoding)(buffer)
+        # StreamWriter does not have encoding attached to it, it will ask
+        # the inner buffer for "encoding" attribute in this case
+        stream.encoding = encoding
+    return stream
+
+
+class TestWriteUnicode(object):
+
+    def setup(self):
+        self.octpus_and_squid_en = u'octpus and squid'
+        octpus_and_squid_zh = u'章鱼和鱿鱼'
+        self.message = self.octpus_and_squid_en + octpus_and_squid_zh
+
+    def test_stdout_writer(self, capsys):
+        # should work with whatever stdout is
+        terminal.stdout(self.message)
+        out, _ = capsys.readouterr()
+        assert self.octpus_and_squid_en in out
+
+    def test_ascii_writer(self, stream, monkeypatch):
+        # should work with stdout with ascii encoding
+        self._test_writer('ascii', stream, monkeypatch)
+
+    def test_utf8_writer(self, stream, monkeypatch):
+        # should work with stdout with UTF-8 encoding
+        self._test_writer('utf8', stream, monkeypatch)
+
+    def _test_writer(self, encoding, stream, monkeypatch):
+        buffer = io.BytesIO()
```

Comment:
buffer is not used by all tests so i leave it in this function.

Reply:
you took what I suggested and made it a thousand times better, this is great work here @tchaikov